### PR TITLE
Cleanups, fixes and a bit of optimizations for site/components batch #5

### PR DESCRIPTION
**Pull request by frankmayer**
- com_tags
- com_users
- com_wrapper

Note: This is a single commit bundling all types of changes, since PR #12261 which had detailed commits, was rejected as a whole

### CODE REVIEW

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on fd62014787712137a987ae0265f57d80a8b473eb<br /><br />on code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12294">issues.joomla.org/joomla-cms/12294</a>.</sub>

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on 62df93ff6db1ecb3459eeac5e100f9264d082136<br /><br />code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12294">issues.joomla.org/tracker/joomla-cms/12294</a>.</sub>

**Comment by frankmayer** [comment]
Fixed some conflicts that had occurred in the meantime.
Having merged the changes of above referenced PR's made this PR lighter and easier to test.
It is essentially ready to be merged, when the team finds the time and PR is OK.

**Review by andrepereiradasilva — APPROVED** [review]

File: components/com_tags/views/tags/view.html.php [hunk 1]
```diff
@@ -166,7 +166,7 @@ protected function _prepareDocument()
 			$this->params->def('page_heading', JText::_('COM_TAGS_DEFAULT_PAGE_TITLE'));
 		}
 
-		if ($menu && ($menu->query['option'] != 'com_tags'))
+		if ($menu && ($menu->query['option'] !== 'com_tags'))
```
Comment (Quy) [inline components/com_tags/views/tags/view.html.php]: Remove parentheses.

File: components/com_tags/views/tag/tmpl/default.php [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 // Note that there are certain parts of this layout used only when there is exactly one tag.
 JHtml::addIncludePath(JPATH_COMPONENT . '/helpers');
-$isSingleTag = (count($this->item) == 1);
+$isSingleTag = (count($this->item) === 1);
```
Comment (Quy) [inline components/com_tags/views/tag/tmpl/default.php]: Remove parentheses

File: components/com_users/router.php [hunk 3]
```diff
@@ -49,37 +49,37 @@ public function build(&$query)
 			for ($i = 0, $n = count($items); $i < $n; $i++)
 			{
 				// Check to see if we have found the resend menu item.
-				if (empty($resend) && !empty($items[$i]->query['view']) && ($items[$i]->query['view'] == 'resend'))
+				if (empty($resend) && !empty($items[$i]->query['view']) && ($items[$i]->query['view'] === 'resend'))
```
Comment (Quy) [inline components/com_users/router.php]: Remove parentheses. `($items[$i]->query['view'] === 'resend')`

File: components/com_users/router.php [hunk 4]
```diff
@@ -49,37 +49,37 @@ public function build(&$query)
...
 				{
 					$resend = $items[$i]->id;
 				}
 
 				// Check to see if we have found the reset menu item.
-				if (empty($reset) && !empty($items[$i]->query['view']) && ($items[$i]->query['view'] == 'reset'))
+				if (empty($reset) && !empty($items[$i]->query['view']) && ($items[$i]->query['view'] === 'reset'))
```
Comment (Quy) [inline components/com_users/router.php]: Remove parentheses.

File: components/com_users/router.php [hunk 5]
```diff
@@ -49,37 +49,37 @@ public function build(&$query)
...
 				{
 					$reset = $items[$i]->id;
 				}
 
 				// Check to see if we have found the remind menu item.
-				if (empty($remind) && !empty($items[$i]->query['view']) && ($items[$i]->query['view'] == 'remind'))
+				if (empty($remind) && !empty($items[$i]->query['view']) && ($items[$i]->query['view'] === 'remind'))
```
Comment (Quy) [inline components/com_users/router.php]: Remove parentheses.

File: components/com_users/router.php [hunk 6]
```diff
@@ -49,37 +49,37 @@ public function build(&$query)
...
 				{
 					$remind = $items[$i]->id;
 				}
 
 				// Check to see if we have found the login menu item.
-				if (empty($login) && !empty($items[$i]->query['view']) && ($items[$i]->query['view'] == 'login'))
+				if (empty($login) && !empty($items[$i]->query['view']) && ($items[$i]->query['view'] === 'login'))
```
Comment (Quy) [inline components/com_users/router.php]: Remove parentheses.

File: components/com_users/router.php [hunk 7]
```diff
@@ -49,37 +49,37 @@ public function build(&$query)
...
 				{
 					$login = $items[$i]->id;
 				}
 
 				// Check to see if we have found the registration menu item.
-				if (empty($registration) && !empty($items[$i]->query['view']) && ($items[$i]->query['view'] == 'registration'))
+				if (empty($registration) && !empty($items[$i]->query['view']) && ($items[$i]->query['view'] === 'registration'))
```
Comment (Quy) [inline components/com_users/router.php]: Remove parentheses.

File: components/com_users/router.php [hunk 8]
```diff
@@ -49,37 +49,37 @@ public function build(&$query)
...
 				{
 					$registration = $items[$i]->id;
 				}
 
 				// Check to see if we have found the profile menu item.
-				if (empty($profile) && !empty($items[$i]->query['view']) && ($items[$i]->query['view'] == 'profile'))
+				if (empty($profile) && !empty($items[$i]->query['view']) && ($items[$i]->query['view'] === 'profile'))
```
Comment (Quy) [inline components/com_users/router.php]: Remove parentheses.

File: components/com_tags/views/tag/tmpl/default_items.php [hunk 9]
```diff
@@ -64,7 +64,7 @@
 	</fieldset>
 	<?php endif; ?>
 
-	<?php if ($this->items == false || $n == 0) : ?>
+	<?php if ($this->items == false || $n === 0) : ?>
```
Comment (Quy) [inline components/com_tags/views/tag/tmpl/default_items.php]: Change `==` to `===`?

File: components/com_tags/views/tag/tmpl/list_items.php [hunk 10]
```diff
@@ -56,7 +56,7 @@
 	</fieldset>
 	<?php endif; ?>
 
-	<?php if ($this->items == false || $n == 0) : ?>
+	<?php if ($this->items == false || $n === 0) : ?>
```
Comment (Quy) [inline components/com_tags/views/tag/tmpl/list_items.php]: Change `==` to `===`?

File: components/com_tags/views/tag/tmpl/default_items.php (hunk 9 above)
Comment (frankmayer) [inline components/com_tags/views/tag/tmpl/default_items.php]: Yes, indeed :+1:

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on bbc3d57d55e8f35a74346e0077af1c9c19aa180a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12294">issues.joomla.org/tracker/joomla-cms/12294</a>.</sub>

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on bbc3d57d55e8f35a74346e0077af1c9c19aa180a<br /><br />On code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12294">issues.joomla.org/tracker/joomla-cms/12294</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Pull request merged**